thunderfish/consistentfishes.py

```diff
@@ -0,0 +1,148 @@
+    :return consistent_fundamentals: (1-D array) List containing all values that are available in all given lists.
+    :return index:          (1-D array) Indices of the values that are in every list relating to the fist list in fishlists.
+    """
+    consistancy_help = [1 for i in np.arange(len(fundamentals[0]))]
+    index = []
+    consistent_fundamentals = []
+
+    for enu, fundamental in enumerate(fundamentals[0]):
+        for list in np.arange(len(fundamentals)-1)+1:
+            freq_diff = [fundamentals[list][i] - fundamental for i in np.arange(len(fundamentals[list]))]
+            for i in np.arange(len(freq_diff)):
+                if freq_diff[i] >= -1.0 and freq_diff[i] <= 1:
+                    consistancy_help[enu] += 1
+                    break
+
+    for idx in np.arange(len(consistancy_help)):
+        if consistancy_help[idx] == len(fundamentals):
+            index.append(idx)
+            consistent_fundamentals.append(fundamentals[0][idx])
+
+    return consistent_fundamentals, index
+
+def consistent_fishlist(index, fishlists):
+    """
+    This function gets the 4-D array of fishlists and the indices of the fishes from the first fishlist that are also
+    available in all other fishlists. It gives back a 3-D array (filterd_fishlist) that only contains the information of
+    the fishes that are in all lists (structur: filtered_fishlist[fish][harmonic][frequency, power])
+
+    :param index:           (1-D array) Indices of the values that are in every list relating to the fist list in fishlists.
+    :param fishlists:       (4-D array) List of fishlists with harmonics and each frequency and power.
+                            fishlists[fishlist][fish][harmonic][frequency, power]
+    :return filtered_fishlist: (3-D array) New fishlist with the same structure as a fishlist in fishlists only
+                            containing these fishes that are available in every fishlist in fishlists.
+                            fishlist[fish][harmonic][frequency, power]
+    """
+    filtered_fishlist = []
+    for idx in index:
+        filtered_fishlist.append(fishlists[0][idx])
+
+    return filtered_fishlist
+
+def consistentfishesplot(filtered_fishlist, ax):
+    """
+    Creates a axis for plotting to visualize what this modul did.
+
+    :param filtered_fishlist: (3-D array) Contains power and frequency of these fishes that hve been detected in
+                            several powerspectrums using different resolutions.
+    :param ax:              (axis for plot) empty axis that is filled with content in the function.
+    :return ax:             (axis for plot) axis that is ready for plotting explaining what the modul did.
+    """
+    for list in np.arange(len(fishlists)):
+        for fish in np.arange(len(fishlists[list])):
+            ax.plot(list+1, fishlists[list][fish][0][0], 'k.', markersize= 10)
+
+    for fish in np.arange(len(filtered_fishlist)):
+        if fish == 0:
```

Comment:
for speeding code, I would put the arguments before the if:
x = np.arange(len(fishlists))+1
y = [filtered_fishlist[fish][0][0] for i in np.arange(len(fishlists))]

then the if-clause.
plot(x,y)